scripts/gen-doc.sh

```diff
@@ -28,39 +28,42 @@ DOCS_REPO="$(dirname "$SOF_REPO")/sof-docs"
 PUBLISH_REPO="$(dirname "$SOF_REPO")/thesofproject.github.io"
 
 # parse arguments
-DO_BUILD=no
-DO_CLEAN=no
-DO_PUBLISH=no
-DO_CLONE=no
-
-for args in $@
-do
-	if [[ "$args" == "-b" ]]
-	then
-		DO_BUILD=yes
-	elif [[ "$args" == "-c" ]]
-	then
-		DO_CLEAN=yes
-	elif [[ "$args" == "-p" ]]
-	then
-		DO_PUBLISH=yes
-	elif [[ "$args" == "-r" ]]
-	then
-		DO_CLONE=yes
-	fi
+DO_BUILD=false
+DO_CLEAN=false
+DO_PUBLISH=false
+FETCH_MISSING=false
+
+while getopts "bcpr" OPTION; do
+        case "$OPTION" in
+        b)
+                DO_BUILD=true
+                ;;
+        c)
+                DO_CLEAN=true
+                ;;
+        p)
+                DO_PUBLISH=true
+                ;;
+        r)
+                FETCH_MISSING=true
```

Comment:
Please call this DO_FETCH for consistency and because `FETCH_MISSING=true` can be misinterpreted as "something related to fetch is missing"